src/services/SolidService.ts

```diff
@@ -99,13 +101,74 @@ class SolidService {
       }
     }
 
-    for (let i = 0; i < info.length; i++) {
-      let title = createThing({ name: `title${i}` });
-      title = addUrl(title, RDF.type, AS.Article);
-      title = addStringNoLocale(title, SCHEMA_INRUPT.name, info[i]);
-      myCovidFile = setThing(myCovidFile, title);
+    let covidTypeThing;
+    let dateThing;
+    let dateUntilThing;
+    switch (certificaat) {
+      case 'vaccinatiecertificaat':
+        covidTypeThing = createThing({ name: `HC1.v` });
+        covidTypeThing = addUrl(
+          covidTypeThing,
+          RDF.type,
+          `http://hl7.eu/fhir/ig/dcc/StructureDefinition/v`,
+        );
+
+        dateThing = createThing({ name: `HC1.v.df` });
+        dateThing = addDate(dateThing, SCHEMA_INRUPT.startDate, dateObj);
+
+        dateUntilThing = createThing({ name: `HC1.v.du` });
+        dateUntilThing = addDate(
+          dateUntilThing,
+          SCHEMA_INRUPT.endDate,
+          validationObj,
+        );
+
+        break;
+      case 'herstelcertificaat':
+        covidTypeThing = createThing({ name: `HC1.r` });
+        covidTypeThing = addUrl(
+          covidTypeThing,
+          RDF.type,
+          `http://hl7.eu/fhir/ig/dcc/StructureDefinition/r`,
+        );
+
+        dateThing = createThing({ name: `HC1.r.df` });
+        dateThing = addDate(dateThing, SCHEMA_INRUPT.startDate, dateObj);
+
+        dateUntilThing = createThing({ name: `HC1.r.du` });
+        dateUntilThing = addDate(
+          dateUntilThing,
+          SCHEMA_INRUPT.endDate,
+          validationObj,
+        );
+        break;
+      case 'testcertificaat':
+        covidTypeThing = createThing({ name: `HC1.t` });
+        covidTypeThing = addUrl(
+          covidTypeThing,
+          RDF.type,
+          `http://hl7.eu/fhir/ig/dcc/StructureDefinition/t`,
+        );
+
+        dateThing = createThing({ name: `HC1.t.df` });
+
+        dateThing = addDate(dateThing, SCHEMA_INRUPT.startDate, dateObj);
+
+        dateUntilThing = createThing({ name: `HC1.t.du` });
+
+        dateUntilThing = addDate(
+          dateUntilThing,
+          SCHEMA_INRUPT.endDate,
+          validationObj,
+        );
+        break;
     }
 
+    const dataItems = [covidTypeThing, dateThing, dateUntilThing];
+
+    for (let i = 0; i < dataItems.length; i++) {
+      myCovidFile = setThing(myCovidFile, dataItems[i]);
+    }
```

Comment:
I would highly recommend using iterators such as `forEach` for these types of loops, they are more readable:

```js
dataItems.forEach((dataItem) => {
  myCovidFile = setThing(myCovidFile, dataItem);
});
```

If you ever come into a situation where you need `i` as a number, you can still get it as the second parameter of the function you pass into the forEach iterator.

```js
dataItems.forEach((dataItem, index) => {
```

For loops can still be used if you need to do different kind of sequences than from `[0, array.length]`, or when comparing/mapping to arrays of different lengths together.

Reply:
Don't forget to also look into:

`array.prototype.map()`
`array.prototype.some()`
`array.prototype.every()`
`array.prototype.reduce()`

Those are some of the functions I use commonly, except for reduce, I don't often use that.